clippy_lints/src/methods/err_expect.rs

```diff
@@ -0,0 +1,54 @@
+use super::ERR_EXPECT;
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::ty::implements_trait;
+use clippy_utils::{meets_msrv, msrvs, ty::is_type_diagnostic_item};
+use rustc_errors::Applicability;
+use rustc_lint::LateContext;
+use rustc_middle::ty;
+use rustc_middle::ty::Ty;
+use rustc_semver::RustcVersion;
+use rustc_span::{sym, Span};
+
+pub(super) fn check(
+    cx: &LateContext<'_>,
+    _expr: &rustc_hir::Expr<'_>,
+    recv: &rustc_hir::Expr<'_>,
+    msrv: Option<&RustcVersion>,
+    expect_span: Span,
+    err_span: Span,
+) {
+    if_chain! {
+        if is_type_diagnostic_item(cx, cx.typeck_results().expr_ty(recv), sym::Result);
+        if meets_msrv(msrv, &msrvs::EXPECT_ERR);
+
+        let result_type = cx.typeck_results().expr_ty(recv);
+        if let Some(error_type) = get_error_type(cx, result_type);
+        if has_debug_impl(error_type, cx);
+
+        then {
+                    span_lint_and_sugg(
+            cx,
+            ERR_EXPECT,
+            err_span.to(expect_span),
+            "called `.err().expect()` on a `Result` value",
+            "try",
+                "expect_err".to_string(),
+                Applicability::MachineApplicable
+        );
```

Comment:
Formatting sadly doesn't work inside `if_chain!` macros. Could you fix the indention? Alternatively, you can use [let chains](https://rust-lang.github.io/rfcs/2497-if-let-chains.html) which should be supported by rustfmt